src/bin/tikv-ctl.rs

```diff
@@ -116,12 +131,149 @@ fn main() {
             }
         }
         dump_range(db, from, to, limit, cf_name);
+    } else if let Some(matches) = matches.subcommand_matches("mvcc") {
+        let cf_name = matches.value_of("cf").unwrap_or("default");
+        let key = matches.value_of("key").unwrap();
+        println!("You are searching Key {}: ", key);
+        match cf_name {
+            CF_DEFAULT => {
+                dump_mvcc_default(db, key);
+            }
+            CF_LOCK => {
+                dump_mvcc_lock(db, key);
+            }
+            CF_WRITE => {
+                dump_mvcc_write(db, key);
+            }
+            _ => {
+                let _ = app.print_help();
+            }
+        }
     } else {
         let _ = app.print_help();
     }
 
 }
 
+fn gen_key_prefix(key: &str) -> Vec<u8> {
+    let mut prefix = keys::DATA_PREFIX_KEY.to_vec();
+    let mut encoded = encode_bytes(key.as_bytes());
+    prefix.append(&mut encoded);
+    prefix
+}
+
+pub struct MVCCDefaultKV {
+    key: Key,
+    value: Vec<u8>,
+}
+
+pub fn gen_mvcc_default_iter(db: &DB, key_prefix: &str) -> Vec<MVCCDefaultKV> {
+    let prefix = gen_key_prefix(key_prefix);
+    let mut iter = db.new_iterator(None);
+    iter.seek(prefix.as_slice().into());
+    if !iter.valid() {
+        println!("No such Default record!");
+        return Vec::new();
+    }
+    iter.map(|s| {
+            let key = &keys::origin_key(&s.0);
+            let len = key.len();
+            let (raw, mut ts) = key.split_at(len - number::U64_SIZE);
+            let mut kv = MVCCDefaultKV {
+                key: Key::from_encoded(raw.to_vec()),
+                value: s.1,
+            };
+            kv.key = kv.key.append_ts(ts.decode_u64_desc().unwrap());
+            kv
+        })
+        .collect()
+}
+
+pub struct MVCCLockKV {
+    key: Key,
+    value: Lock,
+}
+
+pub fn gen_mvcc_lock_iter(db: &DB, key_prefix: &str) -> Vec<MVCCLockKV> {
+    let prefix = gen_key_prefix(key_prefix);
+    let mut iter = db.new_iterator_cf(CF_LOCK, None).unwrap();
+    iter.seek(prefix.as_slice().into());
+    if !iter.valid() {
+        println!("No such Lock record!");
+        return Vec::new();
+    }
+    iter.map(|s| {
+            let key = s.0;
+            MVCCLockKV {
+                key: Key::from_raw(keys::origin_key(&key)),
+                value: Lock::parse(s.1.clone().as_mut_slice()).unwrap(),
+            }
+        })
+        .collect()
+}
+
+pub struct MVCCWriteKV {
+    key: Key,
+    value: Write,
+}
+
+pub fn gen_mvcc_write_iter(db: &DB, key_prefix: &str) -> Vec<MVCCWriteKV> {
```

Comment:
Hmm, they will generate Vector with different types. I think we should separate them.